### PR TITLE
[BNB] Multiple Addresses Tx

### PR DESCRIPTION
**Pull request by Pantani**
# Headline

Add Binance chain support to `Multiple Addresses` transactions for BNB and tokens.

_Example:_
https://explorer.binance.org/tx/F53BB470A3B6B83977CFFE5D5F9937FB1CBB8785FBE818D9B38AD43F3ECD82BC

## Solution

Fetch the detail of all transactions who have the `hasChildren` flag enabled. And parse the subTx as a Tx.

## Checklist

- [x] Fetch tx children
- [x] Normalize children txs
- [x] Add Unit tests
- [x] Test API
- [x] Test observer

## How to test

http://localhost:8420/v1/binance/bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp
```
{
  "total": 2,
  "docs": [
    {
      "id": "4A5E688755E3588B89C5F0325274430AA52786E527313739C542BC69436E18F4",
      "coin": 714,
      "from": "bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp",
      "to": "bnb136ns6lfw4zs5hg4n85vdthaad7hq5m4gtkgf23",
      "fee": "37500",
      "date": 1579689046,
      "block": 63282272,
      "status": "completed",
      "type": "transfer",
      "direction": "outgoing",
      "memo": "102101832",
      "metadata": {
        "value": "55415460",
        "symbol": "BNB",
        "decimals": 8
      }
    },
    {
      "id": "F53BB470A3B6B83977CFFE5D5F9937FB1CBB8785FBE818D9B38AD43F3ECD82BC",
      "coin": 714,
      "from": "bnb1rhv98jcx2yu26shxedskttjzpkvsrz4nd226yv",
      "to": "bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp",
      "fee": "60000",
      "date": 1579688431,
      "block": 63280715,
      "status": "completed",
      "type": "transfer",
      "direction": "incoming",
      "memo": "Trust Wallet Redeem",
      "metadata": {
        "value": "375000",
        "symbol": "BNB",
        "decimals": 8
      }
    }
  ],
  "status": true
}
```

http://localhost:8420/v1/binance/bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp?token=BUSD-BD1
```
{
  "total": 1,
  "docs": [
    {
      "id": "F53BB470A3B6B83977CFFE5D5F9937FB1CBB8785FBE818D9B38AD43F3ECD82BC",
      "coin": 714,
      "from": "bnb1rhv98jcx2yu26shxedskttjzpkvsrz4nd226yv",
      "to": "bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp",
      "fee": "60000",
      "date": 1579688431,
      "block": 63280715,
      "status": "completed",
      "type": "native_token_transfer",
      "direction": "incoming",
      "memo": "Trust Wallet Redeem",
      "metadata": {
        "name": "",
        "symbol": "BUSD",
        "token_id": "BUSD-BD1",
        "decimals": 8,
        "value": "1000000000",
        "from": "bnb1rhv98jcx2yu26shxedskttjzpkvsrz4nd226yv",
        "to": "bnb1563k58pc3keeuwkhlrxwz7sdsetyn9l7gdnznp"
      }
    }
  ],
  "status": true
}
```

closes https://github.com/trustwallet/blockatlas/issues/754

### CODE REVIEW

**Comment by kolya182** [comment]
concept ACK

File: platform/binance/api.go [hunk 1]
```diff
@@ -47,7 +49,11 @@ func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
 	if err != nil {
 		return nil, err
 	}
-	txs := NormalizeTxs(srcTxs.Txs, "")
+	childTxs, err := p.getTxChildChan(srcTxs.Txs)
```
Comment (catenocrypt) [inline platform/binance/api.go]: Naming: what does Chan stands for?

File: platform/binance/model.go [hunk 2]
```diff
@@ -64,6 +65,89 @@ type Tx struct {
...
+	mapTx := map[string]Tx{}
+	for _, subTx := range *subTxs {
+		key := subTx.ToAddr + subTx.Asset
+		tx, ok := mapTx[key]
+		if !ok {
+			mapTx[key] = subTx.toTx()
+			continue
+		}
+		txValue, err := tx.Value.Float64()
+		if err != nil {
```
Comment (dpereskokov) [inline platform/binance/model.go]: do we need this check? default txValue won't be 0?

File: platform/binance/model.go [hunk 3]
```diff
@@ -64,6 +65,89 @@ type Tx struct {
...
+			txValue = 0
+		}
+		subTxValue, err := subTx.Value.Float64()
+		if err != nil {
```
Comment (dpereskokov) [inline platform/binance/model.go]: same thing

File: platform/binance/api.go (hunk 1 above)
Comment (Pantani) [reply platform/binance/api.go]: Channel

File: platform/binance/model.go (hunk 2 above)
Comment (Pantani) [reply platform/binance/model.go]: I can remove it, but I need to see the current implementation of `json.Number` to see if is safety

**Pull request merged**